README.md

```diff
@@ -13,128 +13,339 @@
+| **publish-profile** | The plain-text credentials used to access the `scm` endpoint using HTTP basic authentication during deployment. This must contain the XML contents of your `.PublishSettings` file. T use this authentication method, see [Publish profile (HTTP basic) authentication](#publish-profile-http-basic-authentication). We highly recommend setting the content in GitHub secrets since it contains sensitive information such as your site URL, username, and password. When the publish profile is rotated in your function app, you also need to update the GitHub secret. Otherwise, a 401 error occurs when accessing the /api/settings endpoint. |
+| **respect-pom-xml** | Allow the GitHub Action to derive the Java function app's artifact from pom.xml for deployments. By default, it's set to `false`, which means the **package** parameter needs to point to your Java function app's artifact, such as `./target/azure-functions/<FUNCTION_APP_NAME>`. It's recommended to set **package** to `.` and **respect-pom-xml** to `true` when deploying Java function apps. |
+| **respect-funcignore** | Allow the GitHub Action to parse your .funcignore file and exclude files and folders defined in it. By default, this value is set to `false`. If your GitHub repo contains .funcignore file and you want to exclude certain paths (for example, text editor configs .vscode/, Python virtual environment .venv/), we recommend setting this to `true`. |
 
 ## Dependencies on other GitHub Actions
-* [Checkout](https://github.com/actions/checkout) Checkout your Git repository content into GitHub Actions agent.
-* [Azure Login](https://github.com/Azure/login) Login with your Azure credentials for function app deployment authentication.
-* Environment setup actions
-  * [Setup DotNet](https://github.com/actions/setup-dotnet) Build your DotNet core function app or function app extensions.
-  * [Setup Node](https://github.com/actions/setup-node) Resolve Node function app dependencies using npm.
-  * [Setup Python](https://github.com/actions/setup-python) Resolve Python function app dependencies using pip.
-  * [Setup Java](https://github.com/actions/setup-java) Resolve Java function app dependencies using maven.
 
-# Contributing
++ [Checkout](https://github.com/actions/checkout): Check out your Git repository content into GitHub Actions agent.
++ [Azure Login](https://github.com/Azure/login): Sign in with your Azure credentials when authenticating with OIDC or service principal.
++ Environment setup actions:
+  + [Setup DotNet](https://github.com/actions/setup-dotnet): Build your DotNet core function app or function app extensions.
+  + [Setup Node](https://github.com/actions/setup-node): Resolve Node function app dependencies using npm.
+  + [Setup Python](https://github.com/actions/setup-python): Resolve Python function app dependencies using pip.
+  + [Setup Java](https://github.com/actions/setup-java): Resolve Java function app dependencies using maven.
+
```

Comment:
I realize this just how this section was and maybe outside of the scope of these changes, but should we have any preamble here? These are things you might also need to add in for an E2E, right? Not dependencies we bring in or something.

Reply:
Good point! That would help clarify things.